tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -827,6 +827,7 @@ private void OnHostingUnhandledException(object arg)
 
                 // Generic unhandled exceptions are converted to 500 errors by Kestrel
                 span.SetHttpStatusCode(statusCode: statusCode, isServer: true, tracer.Settings);
+                span.SetHeaderTags(new HeadersCollectionAdapter(unhandledStruct.HttpContext.Response.Headers), tracer.Settings.HeaderTags, defaultTagPrefix: SpanContextPropagator.HttpResponseHeadersTagPrefix);
```

Comment:
Not asking to change anything, more a philosophical question that I asked myself.
I wonder if we could find a way to make sure we ask ourselves the question "should I also set the header" when we set a httpstatuscode. I thought maybe adding a new method on `Span` doing both could be a way as people would see it and ask themselves the question. But I find it kinda ugly. 
Anyway, if you think this comment is relevant and you find a more appropriate idea, it could be nice to reduce the risk of forgetting this in the future.

Reply:
I don't think we can combine these, as we don't set header tags for _client_ requests (e.g. HttpClient, AWS's SDK client etc). We only set them for "server" requests (aspnet, aspnetcore).

If we _should_ be setting them on client http spans, then that's another question! 😄